nav2_behavior_tree/test/plugins/action/test_bt_action_node.cpp

```diff
@@ -483,6 +489,180 @@ TEST_F(BTActionNodeTestFixture, test_server_cancel)
   EXPECT_EQ(ticks, 7);
 }
 
+TEST_F(BTActionNodeTestFixture, test_run_id_initialization_and_persistence)
+{
+  // create tree with is_global="true" to enable RunID checking
+  std::string xml_txt =
+    R"(
+      <root BTCPP_format="4">
+        <BehaviorTree ID="MainTree">
+            <Fibonacci order="100" is_global="true" />
+        </BehaviorTree>
+      </root>)";
+
+  config_->blackboard->set<std::chrono::milliseconds>("server_timeout", 100ms);
+  config_->blackboard->set<std::chrono::milliseconds>("bt_loop_duration", 10ms);
+
+  // Set initial RunID
+  config_->blackboard->set<uint64_t>("run_id", 1);
+
+  tree_ = std::make_shared<BT::Tree>(factory_->createTreeFromText(xml_txt, config_->blackboard));
+  action_server_->setHandleGoalSleepDuration(2ms);
+  action_server_->setServerLoopRate(50ms);  // Slower server rate
+
+  // First tick should initialize with run_id = 1
+  auto result = tree_->tickOnce();
+  EXPECT_EQ(result, BT::NodeStatus::RUNNING);
+
+  // Subsequent ticks with same RunID should continue without re-initialization
+  result = tree_->tickOnce();
+  EXPECT_EQ(result, BT::NodeStatus::RUNNING);
```

Comment:
how come all tests have the same  `EXPECT_EQ(result, BT::NodeStatus::RUNNING);`  ?